test/Microsoft.Framework.PackageManager.FunctionalTests/DnuListTests.cs

```diff
@@ -0,0 +1,144 @@
+            File.WriteAllText(projectJson, @"{}");
+
+            string stdOut, stdErr;
+            Assert.Equal(0, DnuTestUtils.ExecDnu(runtimeHomePath, "list", "", out stdOut, out stdErr, environment: null, workingDir: _workingDir.DirPath));
+        }
+
+        [Theory]
+        [MemberData("RuntimeComponents")]
+        public void DnuList_EmptyProject_Details(string flavor, string os, string architecture)
+        {
+            var runtimeHomePath = _context.GetRuntimeHome(flavor, os, architecture);
+            var projectJson = Path.Combine(_workingDir.DirPath, "project.json");
+            File.WriteAllText(projectJson, @"{}");
+
+            Assert.Equal(0, DnuTestUtils.ExecDnu(runtimeHomePath, "list", "--details", environment: null, workingDir: _workingDir.DirPath));
+        }
+
+        [Theory]
+        [MemberData("RuntimeComponents")]
+        public void DnuList_SingleDependencyProject(string flavor, string os, string architecture)
+        {
+            var runtimeHomePath = _context.GetRuntimeHome(flavor, os, architecture);
+            var projectJson = Path.Combine(_workingDir.DirPath, "project.json");
+            File.WriteAllText(projectJson, @"{
+  ""dependencies"": {
+    ""alpha"": ""0.1.0""
+  },
+  ""frameworks"": {
+    ""dnx451"": {},
+    ""dnxcore50"": {}
+  }
+}");
+
+            Assert.Equal(0, DnuTestUtils.ExecDnu(runtimeHomePath, "restore", "--source " + _context.PackageSource, workingDir: _workingDir.DirPath));
+
+            string stdOut, stdErr;
+            Assert.Equal(0, DnuTestUtils.ExecDnu(runtimeHomePath, "list", "", out stdOut, out stdErr, environment: null, workingDir: _workingDir.DirPath));
+
+            var hits = stdOut.Split('\n').Where(line => line.Contains("* alpha 0.1.0"))
+                                         .Where(line => !line.Contains("Unresolved"));
+            Assert.Equal(2, hits.Count());
+        }
+
+        [Theory]
+        [MemberData("RuntimeComponents")]
+        public void DnuList_SingleDependencyProject_Detailed(string flavor, string os, string architecture)
+        {
+            var runtimeHomePath = _context.GetRuntimeHome(flavor, os, architecture);
+            var projectJson = Path.Combine(_workingDir.DirPath, "project.json");
+            File.WriteAllText(projectJson, @"{
+  ""dependencies"": {
+    ""alpha"": ""0.1.0""
+  },
+  ""frameworks"": {
+    ""dnx451"": {},
+    ""dnxcore50"": {}
+  }
+}");
+
+            var projectName = Path.GetFileName(_workingDir.DirPath);
+
+            Assert.Equal(0, DnuTestUtils.ExecDnu(runtimeHomePath, "restore", "--source " + _context.PackageSource, workingDir: _workingDir.DirPath));
+
+            string stdOut, stdErr;
+            Assert.Equal(0, DnuTestUtils.ExecDnu(runtimeHomePath, "list", "--details", out stdOut, out stdErr, environment: null, workingDir: _workingDir.DirPath));
+
+            string[] outputLines = stdOut.Split(Environment.NewLine[0]);
```

Comment:
`Environment.NewLine[0]` in both situation is `\n`. In the following assertions `string.Contains` is used to examine the content. It is safe.